### PR TITLE
Move callback to the () => {} syntax.

### PR DESCRIPTION
**Pull request by Folyd**
Part of https://github.com/rust-lang/rust/issues/93058.

r? @GuillaumeGomez

### CODE REVIEW

File: src/librustdoc/html/static/js/main.js [hunk 1]
```diff
@@ -57,7 +58,7 @@ function resourcePath(basename, extension) {
     return getVar("root-path") + basename + getVar("resource-suffix") + extension;
 }
 
-(function () {
+((() => {
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/js/main.js:61]: Is it possible to not do it in this case? It's an anonymous function, not a callback after all. ^^'

File: src/librustdoc/html/static/js/main.js [hunk 2]
```diff
@@ -137,7 +138,7 @@ function hideThemeButtonState() {
 }
 
 // Set up the theme picker list.
-(function () {
+((() => {
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/js/main.js:141]: Same (and for the ones below, won't comment on them though).

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me! Please squash your commits (take a look [here](https://blog.guillaume-gomez.fr/Git-tips/1/2) if you don't know how) and then I'll approve it.

**Comment by Folyd** [comment]
> Looks good to me! Please squash your commits (take a look [here](https://blog.guillaume-gomez.fr/Git-tips/1/2) if you don't know how) and then I'll approve it.

Squash done. Thanks, @GuillaumeGomez. 👍

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Pull request merged**